### DISCUSSION TITLE
Auth0 implementation: not working permanently

### DISCUSSION THREAD

**Discussion by nbolution** [strapi/strapi#6114] [Legacy Topics]
Anybody there who works with Auth0?
I try since a couple of days to build a setup with Auth0 to get an access token, which can make secure endpoint-requests.
It works!! (this is an important detail for the issue). But not permanently (At same time). I get my access token from Auth0 well. And sometimes my calls work with that token(s) (send via Autorization Bearer to strapi endpoints) and sometimes they won't. If I use the tokens genarted by Strapi himself the calls works permanently fine. Don't know what I'm doing wrong.
I have followed the guide in documentation to customize the permissions.js. And again, config seems to be right, because sometimes identically calls work and in the next second they don't.
Interesting: I use Angular, and I would like to get some start configs for the app, from the api. When I start the app, first time calls mostly works correct. If I reload the app or navigate to another route which also needs data from the api (Token still valid) some api calls works and others don't (with same config via permissions in the Admin panel for the ressources).

Really confusing. Has anybody an idea or had an similiar behavior? Should I send more Information pls let me know. I'm new to strapi and this is my first publish question here :-)
Want to say at the end of this post: I really loves Strapi! Very great work! :-)

Using strapi-beta-20.1

**Comment by lauriejim** [comment]
Hello! can you please attach the guide you followed.

After that, if the token is well received in the strapi app and the request to Auth0 is done with the right token, the issue should come from the Auth0 side I think.

Because Strapi has no impact in the token validation since you implemented it by yourself (I mean you wrote the code that check the token on Auth0)
